libs/pyrt_component/main.py

```diff
@@ -0,0 +1,58 @@
+import pyrt.exports
+
+# NOTE: all imports must be toplevel as constrained by `componentize-py`
+from pyrt.imports.shared import Req, Ok, Err
+
+# from pyrt.imports.typegate_wire import hostcall
+from pyrt.exports.mat_wire import (
+    InitArgs,
+    InitResponse,
+    InitError_UnexpectedMat,
+    MatInfo,
+)
+
+import json
+import types
+from typing import Callable, Any
+
+handlers = {}
+
+
+class MatWire(pyrt.exports.MatWire):
+    def init(self, args: InitArgs):
+        for op in args.expected_ops:
+            handlers[op.op_name] = op_to_handler(op)
+        return InitResponse(ok=True)
+
+    def handle(self, req: Req):
+        handler = handlers.get(req.op_name)
+        if handler is None:
+            return Err(json.dumps({"ty": "handler_404", "op_name": req.op_name}))
+        try:
+            return handler.handle(req)
+        except Exception as err:
+            return Err(json.dumps({"ty": "handler_500", "message": str(err)}))
+
+
+class ErasedHandler:
+    def __init__(self, handler_fn: Callable[[Any], Any]) -> None:
+        self.handler_fn = handler_fn
+
+    def handle(self, req: Req):
+        in_parsed = json.loads(req.in_json)
+        out = self.handler_fn(in_parsed)
+        return Ok(json.dumps(out))
+
+
+def op_to_handler(op: MatInfo) -> ErasedHandler:
```

Comment:
Importing modules is yet pending. It's a bit tricky here but using importlib, it should be doable.